net/dnscache/dnscache.go

```diff
@@ -457,9 +457,7 @@ func TLSDialer(fwd DialContextFunc, dnsCache *Resolver, tlsConfigBase *tls.Confi
 			}
 		}()
 		go func() {
-			err := tlsConn.Handshake()
-			handshakeTimeoutCancel()
-			errc <- err
+			errc <- tlsConn.Handshake()
```

Comment:
really all this crap can be simplified now that this is in std as of Go 1.17: https://pkg.go.dev/crypto/tls#Conn.HandshakeContext

Reply:
Replaced with HandshakeContext.